### PR TITLE
Ensure that deleted_at check is included only once

### PR DESCRIPTION
**Pull request by ayan-haider**
This PR resolves the following issue:
https://github.com/revelrylabs/ecto_soft_delete/issues/159

### CODE REVIEW

File: test/soft_delete_repo_test.exs [hunk 1]
```diff
@@ -136,5 +136,27 @@ defmodule Ecto.SoftDelete.Repo.Test do
 
       assert length(results) == 1
     end
+
+    test "returns same result for different types of where clauses", %{topic: topic} do
```
Comment (quinnmccourt) [inline test/soft_delete_repo_test.exs]:
```suggestion
    test "returns same result for different types of where clauses" do
```
the second argument `%{topic: topic}` is causing the test to fail.

**Review by quinnmccourt — CHANGES_REQUESTED** [review]
Thank you for your contribution! I have suggested a change that should fix the test failure.

File: test/soft_delete_repo_test.exs (hunk 1 above)
Comment (ayan-haider) [reply test/soft_delete_repo_test.exs]: Oh, thank you for pointing it out. 
Was checking something out before pushing my changes and forgot to remove this. :sweat_smile:

**Review by quinnmccourt — APPROVED** [review]

**Review by stuartjohnpage — APPROVED** [review]
This looks like a good change to me!

**Comment by stuartjohnpage** [comment]
NBD - I will say, looking at the documentation for `inspect()` it seems that [Inspect.Algebra.to_doc/2](https://hexdocs.pm/elixir/1.9.4/Inspect.Algebra.html#to_doc/2) is [preferred](https://hexdocs.pm/elixir/1.9.4/Inspect.html#inspect/2) to just using a raw `inspect()`

**Comment by ayan-haider** [comment]
> NBD - I will say, looking at the documentation for `inspect()` it seems that [Inspect.Algebra.to_doc/2](https://hexdocs.pm/elixir/1.9.4/Inspect.Algebra.html#to_doc/2) is [preferred](https://hexdocs.pm/elixir/1.9.4/Inspect.html#inspect/2) to just using a raw `inspect()`

Thanks for pointing that out @stuartjohnpage, I did want to use that initially but I noticed that the doc returned is in some other format. (I've attached it below for reference). I thought maybe it might overcomplicate a relatively simple task but I'll look more into it :smile: .

```
{:doc_group,
 {:doc_cons,
  {:doc_cons, "{",
   {:doc_nest,
    {:doc_cons, {:doc_cons, ":is_nil", ","},
     {:doc_cons, {:doc_break, " ", :flex},
      {:doc_cons, {:doc_cons, "[]", ","},
       {:doc_cons, {:doc_break, " ", :flex},
        {:doc_group,
         {:doc_cons,
          {:doc_nest,
           {:doc_cons, "[",
            {:doc_cons, {:doc_break, "", :strict},
             {:doc_group,
              {:doc_cons,
               {:doc_cons, "{",
                {:doc_nest,
                 {:doc_cons,
                  {:doc_cons,
                   {:doc_group,
                    {:doc_cons,
                     {:doc_cons, "{",
                      {:doc_nest,
                       {:doc_cons, {:doc_cons, ":.", ","},
                        {:doc_cons, {:doc_break, " ", :flex},
                         {:doc_cons, {:doc_cons, "[]", ","},
                          {:doc_cons, {:doc_break, " ", :flex},
                           {:doc_group,
                            {:doc_cons,
                             {:doc_nest,
                              {:doc_cons, "[",
                               {:doc_cons, {:doc_break, "", :strict},
                                {:doc_cons,
                                 {:doc_cons,
                                  {:doc_group,
                                   {:doc_cons,
                                    {:doc_cons, "{",
                                     {:doc_nest,
                                      {:doc_cons, {:doc_cons, ":&", ","},
                                       {:doc_cons, {:doc_break, " ", :flex},
                                        {:doc_cons, {:doc_cons, "[]", ","},
                                         {:doc_cons, {:doc_break, " ", :flex},
                                          {:doc_group,
                                           {:doc_cons,
                                            {:doc_cons, "[",
                                             {:doc_nest, "0", 1, :always}},
                                            "]"}, :self}}}}}, 1, :always}},
                                    "}"}, :self}, ","},
                                 {:doc_cons, {:doc_break, " ", :strict},
                                  ":deleted_at"}}}}, 2, :always},
                             {:doc_cons, {:doc_break, "", :strict}, "]"}},
                            :self}}}}}, 1, :always}}, "}"}, :self}, ","},
                  {:doc_cons, {:doc_break, " ", :flex},
                   {:doc_cons, {:doc_cons, "[]", ","},
                    {:doc_cons, {:doc_break, " ", :flex}, "[]"}}}}, 1,
                 :always}}, "}"}, :self}}}, 2, :always},
          {:doc_cons, {:doc_break, "", :strict}, "]"}}, :self}}}}}, 1,
    :always}}, "}"}, :self}

**Comment by ayan-haider** [comment]
Maybe we could do something like this but I wasn't sure about it:

`Inspect.Algebra.to_doc(expr, %Inspect.Opts{limit: :infinity, inspect_fun: fn expr, _ -> inspect(expr) end})`

Also, I think this is how they want it to be used as mentioned in the docs: 

"This function shouldn't be invoked directly, unless when implementing a custom `inspect_fun` to be given to [Inspect.Opts](https://hexdocs.pm/elixir/1.9.4/Inspect.Opts.html). Everywhere else, [Inspect.Algebra.to_doc/2](https://hexdocs.pm/elixir/1.9.4/Inspect.Algebra.html#to_doc/2) should be preferred as it handles structs and exceptions."

**Comment by stuartjohnpage** [comment]
@ayan-haider yes, that looks like what we want!

Thank you for taking the time to do that!

**Comment by ayan-haider** [comment]
@quinnmccourt @stuartjohnpage, is this PR good to go as all checks are green?

**Comment by stuartjohnpage** [comment]
@ayan-haider yup!

**Review by stuartjohnpage — APPROVED** [review]

**Comment by mubeenghauri** [comment]
Hey guys, 
This looks like a good change, I was also hoping to fix this, found the issue and this PR.
Should this be merged?

**Comment by ayan-haider** [comment]
@quinnmccourt Could you please merge this?

**Pull request merged**